### DISCUSSION TITLE
Eslint rule/plugin for detecting whitespace text nodes in JSX

### DISCUSSION THREAD

**Discussion by zanlib0** [eslint/eslint#16595] [Q&A]
Is there an ESlint plugin that warns you when you try to do something like this?

```javascript
const Foo = () => (
  <Bar> {arg => <div>{arg}</div>}</Bar>
)
```

If `Bar` expects children to be a function, this will cause an error since the space before the expression results in React interpreting it as a text node and Bar will throw `children is not a function` error. It's pretty difficult to pick up if you're not sure that you're looking specifically for that.

**Comment by mdjermanovic** [comment]
Check [`eslint-plugin-react` rules](https://github.com/jsx-eslint/eslint-plugin-react#list-of-supported-rules). If there's no such rule there, you can probably configure [`no-restrict-syntax`](https://eslint.org/docs/latest/rules/no-restricted-syntax) to catch this. Here's an example selector that might be helpful:

```
selector: "JSXElement[openingElement.name.name='Bar'] > JSXText[value=/^\\s+$/].children"
```

[Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJlc3RyaWN0ZWQtc3ludGF4OiBbXCJlcnJvclwiLCB7XG4gICAgc2VsZWN0b3I6IFwiSlNYRWxlbWVudFtvcGVuaW5nRWxlbWVudC5uYW1lLm5hbWU9J0JhciddID4gSlNYVGV4dFt2YWx1ZT0vXlxcXFxzKyQvXS5jaGlsZHJlblwiLFxuICAgIG1lc3NhZ2U6IFwiVW5leHBlY3RlZCB3aGl0ZXNwYWNlIGluIDxCYXI+LlwiXG59XSovXG5cbmNvbnN0IEZvbyA9ICgpID0+IChcbiAgPEJhcj4ge2FyZyA9PiA8ZGl2Pnthcmd9PC9kaXY+fTwvQmFyPlxuKSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfX0sInJ1bGVzIjp7fSwiZW52Ijp7ImVzNiI6dHJ1ZX19fQ==)
